libraries/AP_Logger/LoggerMessageWriter.cpp

```diff
@@ -233,14 +233,28 @@ void LoggerMessageWriter_WriteSysInfo::process() {
         stage = Stage::RC_PROTOCOL;
         FALLTHROUGH;
 
-    case Stage::RC_PROTOCOL:
+    case Stage::RC_PROTOCOL: {
         const char *prot = hal.rcin->protocol();
         if (prot == nullptr) {
             prot = "None";
         }
         if (! _logger_backend->Write_MessageF("RC Protocol: %s", prot)) {
             return; // call me again
         }
+        stage = Stage::BATT_CYCLES;
+        FALLTHROUGH;
+    }
+    case Stage::BATT_CYCLES:
+    
+        const AP_BattMonitor &battery = AP::battery();
+        for (uint8_t i = 0; i < AP_BATT_MONITOR_MAX_INSTANCES; i++) {
+            uint16_t cycles = 0;
+            if(battery.get_cycle_count(i, cycles)) {
+                if (! _logger_backend->Write_MessageF("BATT%u_MONITOR: Cycles %u", i + 1, cycles)) {
```

Comment:
Really thinking we should formalise this information into a real message.  We don't really want people writing log analysis software which is pulling apart statustexts if we can avoid it.

Writing these out isn't hard - its in the Wiki and an example is here: https://github.com/ardupilot/ardupilot/blob/master/libraries/AP_TECS/AP_TECS.cpp#L1191

See `README.md` in the `AP_Logger` directory for more information, too.